dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1031,11 +1032,19 @@ BlockInputStreams DeltaMergeStore::readRaw(const Context & db_context,
                 auto segment_snap = segment->createSnapshot(*dm_context, false, CurrentMetrics::DT_SnapshotOfReadRaw);
                 if (unlikely(!segment_snap))
                     throw Exception("Failed to get segment snap", ErrorCodes::LOGICAL_ERROR);
+
                 tasks.push_back(std::make_shared<SegmentReadTask>(segment, segment_snap, RowKeyRanges{segment->getRowKeyRange()}));
             }
         }
     }
 
+    fiu_do_on(FailPoints::force_slow_page_storage_snapshot_release, {
+        std::thread thread_hold_snapshots([tasks]() {
+            std::this_thread::sleep_for(std::chrono::seconds(5 * 60));
+            (void)tasks;
+        });
```

Comment:
This creates a std::thread object without calling `deatch`/`join`, will it make TiFlash crash if we enable this failpoint?
https://en.cppreference.com/w/cpp/thread/thread/~thread

Should we call `deatch` on the thread object?

Reply:
oh, it is diff with `pthread`. added.